### PR TITLE
Task/i2c slave driver issue 90

### PR DESCRIPTION
**Pull request by ToddG**
For discussion only. No actual merge requested.

### CODE REVIEW

File: avr-hal-generic/src/i2c.rs [hunk 1]
```diff
@@ -499,5 +499,82 @@ macro_rules! impl_twi_i2c {
                 Ok(())
             }
         }
+
+        $(#[$i2c_attr])*
+        pub struct Slave<CLOCK: $crate::clock::Clock, M> {
```
Comment (ToddG) [inline avr-hal-generic/src/i2c.rs:504]: Need to call this something besides `Slave`. You had suggested `$I2cSlave` but that caused errors in the macro expansion or subsequent build.

File: avr-hal-generic/src/i2c.rs [hunk 2]
```diff
@@ -499,5 +499,82 @@ macro_rules! impl_twi_i2c {
...
+            /// TODO: verify whether we need the speed param (bitrate)
+            pub fn new(
+                p: $I2C,
+                sda: $sdamod::$SDA<$crate::port::mode::Input<$crate::port::mode::PullUp>>,
+                scl: $sclmod::$SCL<$crate::port::mode::Input<$crate::port::mode::PullUp>>,
+                speed: u32,
+                address: u8,
+            ) -> Slave<CLOCK, $crate::i2c::I2cPullUp> {
+                // Calculate TWBR
+                let twbr = ((CLOCK::FREQ / speed) - 16) / 2;
```
Comment (ToddG) [inline avr-hal-generic/src/i2c.rs:531]: Where do I need to look (study) to see what TWBR is and whether a `slave` needs it configured?

**Comment by Rahix** [comment]
> For discussion only. No actual merge requested.

You can mark a PR as a draft to prevent it being merged in its current state.  Mind doing that? :)

**Comment by ToddG** [comment]
Sure thing. I'll close this and make a new `draft` pr. Didn't know GH supported that. Cool.

**Comment by Rahix** [comment]
Nono, mark this one as draft, you can change it!

**Comment by ToddG** [comment]
Oops. I didn't see that as an option.

-Todd

‐‐‐‐‐‐‐ Original Message ‐‐‐‐‐‐‐

**Comment by Rahix** [comment]
Never mind, let's move over to the other PR.

**Pull request closed without merging**